x-pack/plugins/osquery/server/routes/live_query/create_live_query_route.ts

```diff
@@ -113,8 +113,9 @@ export const createLiveQueryRoute = (router: IRouter, osqueryContext: OsqueryApp
           body: { data: osqueryAction },
         });
       } catch (error) {
-        // TODO validate for 400 (when agents are not found for selection)
-        // return response.badRequest({ body: new Error('No agents found for selection') });
+        if (error.statusCode === 400) {
```

Comment:
What about checking for the `error.message === No agents found for selection` so you avoid creating a new type and manipulating the error object. If not, you could create a custom error class so you don't have to manipulate the `Error` one. Thoughts?